### DISCUSSION TITLE
@apply 无法嵌套

### DISCUSSION THREAD

**Discussion by vinciarts** [unocss/unocss#3982] [General]
我在 app.css 里想嵌套 @apply。但只有第一层可以。
例如这样可以正常运行：
```
.btn {
  @apply flex px-3
}
```
但是这样就不行了：
```
.btn_layout {
  @apply flex px-3
}
.btn {
  @apply btn_layout
}
```

是我什么设置问题吗？或者有什么等价替代方案？以前用 tailwind 的时候这样是可以正常工作的。

**Comment by s3xysteak** [comment]
你可能需要在uno.config.ts的 [shortcut](https://unocss.dev/config/shortcuts) 里加入，然后使用shortcut作为简写

**Reply by vinciarts** [reply]
可以了，感谢！顺便问个扩展问题，shortcut会把这些 class 里的样式合并在一个里。能不能分开啊，比如
```
btn: 'a b c'
```
我希望最后还是在元素上增加 `a` `b` `c` 三个 class。这样感觉最后生成的样式会更加精简点 (吧)

**Reply by s3xysteak** [reply]
> 可以了，感谢！顺便问个扩展问题，shortcut会把这些 class 里的样式合并在一个里。能不能分开啊，比如
> 
> ```
> btn: 'a b c'
> ```
> 
> 我希望最后还是在元素上增加 `a` `b` `c` 三个 class。这样感觉最后生成的样式会更加精简点 (吧)

是指这样吗 

```js
shortcuts: {
  'btn': 'a1 a2 b1 b2 c1 c2',
}
<button btn />
```

↓

```js
shortcuts: {
  'a': 'a1 a2',
  'b': 'b1 b2',
  'c': 'c1 c2'
}
<button a b c />
```

**Reply by vinciarts** [reply]
好像不太是…

举个例子，如果我这么写 `<button class="button">`，
根据现在的 shortcuts ，你会得到一个 `<button class="button">`。`.button` 里包含了 `.a` `.b` `.c` 的样式。
但我想渲染出来的是 `<button class="a b c">`。因为 `.a` `.b` `.c` 这几个 class 会用在很多地方，我想不产生重复样式。

也就是 shortcuts 直接把我定义的 class 成员照搬进去就可以。除非那个成员也是一个是 shortcuts，那就继续拆

**Reply by s3xysteak** [reply]
[或许你是想这样？](https://unocss.dev/play/?html=DwIwrgLhD2B2AEIKwHxNsA9OKcUChRIYEBDFUrHEgo3BENK4vIA&config=PTAEFMGcBsEsDsAuBaAJrSBDARtcz5wAPFOQ0eAe2QCcpEbYBjRcVZWAWwAdKbFIAKC69%2BoAN6DQoVOABmCcAGFK8BQHMANFNDc6kcIgCCiBrGwBXRLDkBPbdL1RDASSarID3fsMBVKtoAvqByNJScoADkFlRMkJCRgoLEoogy8pgW0GmyCoQqarDqABSS0pAAFnyITFaQAFwSOtKRmJGNkUzIuBbgkV4t2O1R2OrI3AgA1v3NUUzDkQDuyACMAAygFcgALGsz0oOI8AuYoEM6gV5OBgKNANqz135UxQCUA97OxqaMltZ2by8AF0gq9BEA&css=PQKgBA6gTglgLgUzAYwK4Gc4HsC2YDCAyoWABYJQIA0YAhgHYAmYcUD6AZllDhWOqgAOg7nAB0YAGLcwCAB60cggDYIAXGBDAAUKDBi0mXGADe2sGC704AWgDuCGAHNScDQFYADJ4Dc5sAACtMLKAJ5gggCMLPK2ABR2pPBIcsoAlH4WAEa0yADWTlBYqEw2yFjK3Bpw5LxxAOTllVDoYpSMYgAs3vUZ2gC%2BmsBAA&options=N4IgLgTghgdgzgMwPYQLYAkyoDYgFwJTZwCmAvkA)

一方面，你可以把a b c单独写成shortcuts，另一方面shortcuts的内容也可以是shortcuts，比如示例中的'btn': 'a b c'

如果你是从生产环境output的角度来说，把btn替换为 a b c 估计不好整，那样的话可能得写 [完全自定义的rules](https://unocss.dev/config/rules)

**Reply by vinciarts** [reply]
了解了，感谢解答，那我先忍一忍就多一些 class 吧…自己不想折腾哈哈
